components/ConnectLedgerDialogContent/index.tsx

```diff
@@ -47,6 +79,7 @@ const ConnectLedgerDialogContent: React.FC<ConnectLedgerDialogContentProps> = ({
 
   React.useEffect(() => {
     connectLedger()
+    signInstructions.forEach((item, i) => setTimeout(() => setInstruction(item), (i + 1) * 5000))
```

Comment:
are you sure we are using timeout to rotate the instruction instead of detecting the ledger state?

Reply:
btw this line here will cause memory leak (try closing the screen before all instructions are loaded, then check the console) so i suggest using `setInterval` and `clearInterval` on unmount

Reply:
or it might be better to use a carousel library like this one
https://www.npmjs.com/package/react-material-ui-carousel